experimental/examples/imageviewer/shared/src/androidMain/kotlin/example/imageviewer/view/CameraView.android.kt

```diff
@@ -1,21 +1,155 @@
 
+private val executor = Executors.newSingleThreadExecutor()
+
+@OptIn(ExperimentalPermissionsApi::class)
+@Composable
+internal actual fun CameraView(
+    modifier: Modifier,
+    onCapture: (picture: PictureData.Camera, image: PlatformStorableImage) -> Unit
+) {
+    val cameraPermissionState = rememberMultiplePermissionsState(
+        listOf(
+            android.Manifest.permission.CAMERA,
+            android.Manifest.permission.ACCESS_COARSE_LOCATION,
+            android.Manifest.permission.ACCESS_FINE_LOCATION,
+        )
+    )
+    if (cameraPermissionState.allPermissionsGranted) {
+        CameraWithGrantedPermission(modifier, onCapture)
+    } else {
+        LaunchedEffect(Unit) {
+            cameraPermissionState.launchMultiplePermissionRequest()
+        }
+    }
+}
+
+@SuppressLint("MissingPermission")
 @Composable
-internal actual fun CameraView(modifier: Modifier) {
-    Box(Modifier.fillMaxSize().background(Color.Black)) {
-        Text(
-            text = "Camera is not available on Android for now.",
-            color = Color.White,
-            modifier = Modifier.align(Alignment.Center)
+private fun CameraWithGrantedPermission(
+    modifier: Modifier,
+    onCapture: (picture: PictureData.Camera, image: PlatformStorableImage) -> Unit
+) {
+    val context = LocalContext.current
+    val lifecycleOwner = LocalLifecycleOwner.current
+
+    val preview = Preview.Builder().build()
+    val previewView = remember { PreviewView(context) }
+    val imageCapture: ImageCapture = remember { ImageCapture.Builder().build() }
+    val cameraSelector = CameraSelector.Builder()
+        .requireLensFacing(CameraSelector.LENS_FACING_BACK)
+        .build()
+
+    LaunchedEffect(Unit) {
+        val cameraProvider = suspendCoroutine<ProcessCameraProvider> { continuation ->
+            ProcessCameraProvider.getInstance(context).also { cameraProvider ->
+                cameraProvider.addListener({
+                    continuation.resume(cameraProvider.get())
+                }, executor)
+            }
+        }
+        cameraProvider.unbindAll()
+        cameraProvider.bindToLifecycle(
+            lifecycleOwner,
+            cameraSelector,
+            preview,
+            imageCapture
         )
+        preview.setSurfaceProvider(previewView.surfaceProvider)
+    }
+    val nameAndDescription = createNewPhotoNameAndDescription()
+    var capturePhotoStarted by remember { mutableStateOf(false) }
+    Box(contentAlignment = Alignment.BottomCenter, modifier = modifier) {
+        AndroidView({ previewView }, modifier = Modifier.fillMaxSize())
+        Button(
+            enabled = !capturePhotoStarted,
+            onClick = {
+                capturePhotoStarted = true
+                imageCapture.takePicture(executor, object : OnImageCapturedCallback() {
+                    override fun onCaptureSuccess(image: ImageProxy) {
+                        val byteArray: ByteArray = image.planes[0].buffer.toByteArray()
```

Comment:
Mix of low-level and UI in the same function doesn't look good. I'd suggest extract low-level implementation into helper function (with callback or so)

Reply:
Yes, but for now it is simpler, because it is used only in one place